### DISCUSSION TITLE
Equinox Launcher FIle Locking on Windows

### DISCUSSION THREAD

**Discussion by kigero** [eclipse-equinox/equinox#679] [Q&A] [answered]
First off, I get that this question is a bit of a long shot, but I'm getting nowhere myself so thought I'd ask.

I'm trying to resolve an issue that I'm having with the VSCode Java extensions, which seem to be using the equinox launcher to handle (at least some) of its functionality.  I've got some details in a post [here](https://github.com/redhat-developer/vscode-java/issues/3512), but the summary is that when I've got a java project opened in VSCode (on Windows), there seems to be a lock of some sort being held on the JAR files referenced in a POM.xml file.  This results in those referenced JAR files not being able to be rebuilt without shutting down the VSCode java extension (which closes the equinox launcher), doing the rebuild, and then starting the extension back up.  If I run the exact same project in Eclipse directly, I don't see this behavior.

I've spent a little bit of time looking through the equinox code, but it's a massive code base and I haven't been able to find a good entry point (or even get it compiled yet), so I thought I'd ask whether there are any pointers to tracking this down - or if it's expected, or just obvious that it's not an equinox problem but a VSCode problem.

Thanks!

**Comment by tjwatson** [comment, marked as answer]
There is no evidence here that Equinox code is holding locks on dependency JARs.  This is something [jdt.ls](https://github.com/eclipse-jdtls/eclipse.jdt.ls) should have a look at.  Seems similar to https://github.com/eclipse-jdt/eclipse.jdt.core/issues/1211

**Comment by kigero** [comment]
Sounds reasonable - and that issue seems much closer to the culprit, so I'm glad I asked, now I have another avenue to follow up on.  Thanks!
